### PR TITLE
Refactor `update-template-ubuntu.sh` and add `update-template-debian.sh`

### PR DESCRIPTION
**Pull request by norio-nomura**
- [x] Refactor `update-template-ubuntu.sh` to separate distribution-specific code.
- [x] Enable support for other distributions.
- [x] Add a common `update-template.sh`.
- [x] Add `update-template-debian.sh` to handle Debian.

This PR aims to add Debian support as per #1347.
Also fixes: #2756

### CODE REVIEW

**Comment by norio-nomura** [comment]
Finally, I've laid the groundwork to add code for other distributions. Now, I'll start adding code for Debian.

**Comment by norio-nomura** [comment]
Added `hack/update-template-debian.sh`:
```console
$ hack/update-template-debian.sh
update-template-debian.sh: Update the Debian image location in the specified templates

Usage:
  update-template-debian.sh [--backports[=<bool>]] [--daily[=<bool>]] [--timestamped[=<bool>]] [--version <version>] <template.yaml>...

Description:
  This script updates the Debian image location in the specified templates.
  If the image location in the template contains a release date in the URL, the script replaces it with the latest available date.
  If no flags are specified, the script uses the version from the image location basename in the template.

  Image location basename format: debian-<version>[-backports]-genericcloud-<arch>[-daily][-<timestamp>].qcow[2](https://github.com/lima-vm/lima/issues/2)

  Published Debian image information is fetched from the following URLs:

    https://cloud.debian.org/images/cloud/<codename>[-backports]/[daily/](latest|<timestamp>)/debian-<version>[-backports]-genericcloud-<arch>[-daily][-<timestamp>].json

  The downloaded JSON file will be cached in the Lima cache directory.

Examples:
  Update the Debian image location in templates/**.yaml:
  $ update-template-debian.sh templates/**.yaml

  Update the Debian image location in ~/.lima/debian/lima.yaml:
  $ update-template-debian.sh ~/.lima/debian/lima.yaml

  Update the Debian image location to debian-[13](https://github.com/lima-vm/lima/issues/13)-genericcloud-<arch>.qcow[2](https://github.com/lima-vm/lima/issues/2) in ~/.lima/debian/lima.yaml:
  $ update-template-debian.sh --version trixie ~/.lima/debian/lima.yaml

Flags:
  --backports[=<bool>]    Use the backports image
                          The boolean value can be true, false, [1](https://github.com/lima-vm/lima/issues/1), or [0](https://github.com/lima-vm/lima/issues/0)
  --daily[=<bool>]        Use the daily image
  --timestamped[=<bool>]  Use the timestamped image
  --version <version>     Use the specified version
                          The version can be a codename, version number, or alias (testing, stable, oldstable)
  -h, --help              Print this help message
```

**Comment by norio-nomura** [comment]
The `update-template.sh` script updates the date-based URLs for all supported distributions.  
The `update-template-<distribution>.sh` scripts allow distribution-specific options to be used.

**Comment by AkihiroSuda** [comment]
Thanks, I tried `./hack/update-template.sh templates/*.yaml` but it didn't update Debian

```console
$ ./hack/update-template.sh templates/*.yaml
[...]
Processing templates/centos-stream.yaml                                                                                         
Processing templates/debian-11.yaml                                                                                             
Processing templates/debian-12.yaml                                                                                             
Processing templates/debian.yaml                                                                                                
Processing templates/default.yaml                                                                                               
{                                                                                                                               
  "location": "https://cloud-images.ubuntu.com/releases/24.04/release-20241004/ubuntu-24.04-server-cloudimg-amd64.img",         
  "arch": "x86_64",                                             
  "digest": "sha256:fad101d50b06b26590cf30542349f9e9d3041ad7929e3bc3531c81ec27f2c788"                                           
}
...
```

File: hack/update-template-debian.sh [hunk 1]
```diff
@@ -0,0 +1,435 @@
...
+
+Examples:
+  Update the Debian image location in templates/**.yaml:
+  $ $(basename "${BASH_SOURCE[0]}") templates/**.yaml
+
+  Update the Debian image location in ~/.lima/debian/lima.yaml:
+  $ $(basename "${BASH_SOURCE[0]}") ~/.lima/debian/lima.yaml
+
+  Update the Debian image location to debian-13-genericcloud-<arch>.qcow2 in ~/.lima/debian/lima.yaml:
+  $ $(basename "${BASH_SOURCE[0]}") --version trixie ~/.lima/debian/lima.yaml
```
Comment (AkihiroSuda) [inline hack/update-template-debian.sh:40]: This needs `limactl refactor-reset`

File: hack/update-template.sh [hunk 2]
```diff
@@ -0,0 +1,234 @@
...
+	#     - Returns: cache_key (string)
+	#     - Exits with an error if the image location is not supported.
+	#   - ${distribution}_image_entry_for_image_kernel
+	#     - Arguments: location, kernel_location
+	#     - Returns: image_entry (JSON object)
+	#	  - Exits with an error if the image location is not supported.
+	declare -a SUPPORTED_DISTRIBUTIONS=()
+
+	# shellcheck source=./update-template-ubuntu.sh
+	. "${scriptdir}/update-template-ubuntu.sh"
```
Comment (AkihiroSuda) [inline hack/update-template.sh:147]: Probably you forgot to add Debian here?

Comment (norio-nomura) [reply hack/update-template.sh:147]: Yup, fixed. 👍🏻

File: hack/update-template-debian.sh (hunk 1 above)
Comment (norio-nomura) [reply hack/update-template-debian.sh:41]: added
````
  $ limactl factory-reset debian
```
to help

**Comment by AkihiroSuda** [comment]
Please consider squashing the commits

**Comment by norio-nomura** [comment]
> Please consider squashing the commits

done.

**Review by AkihiroSuda — APPROVED** [review]
Thanks

**Comment by norio-nomura** [comment]
Thanks! 🙏🏻

**Pull request merged**